### PR TITLE
use os.environ to define app_ver

### PR DESCRIPTION
**Pull request by CaiMiao**
I haven't set condition if app_ver is not defined

### CODE REVIEW

**Comment by summertriangle-dev** [comment]
Can you tag the variable name with `VC_` so it's like the other updating ones? Also if you pass a second value to os.environ.get it will be used as a fallback.

File: starlight/__init__.py [hunk 1]
```diff
@@ -400,6 +400,12 @@ def check_version():
             return
 
         print("trace check_version")
+        if ( os.environ.get("VC_APP_VER") == None ):
```
Comment (summertriangle-dev) [inline starlight/__init__.py]: This check should go in `can_check_version()`.

File: starlight/apiclient.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def call(self, path, args, done):
             "USER_ID": self.lolfuscate(str(self.user)),
             "CARRIER": "google",
             "UDID": self.lolfuscate(self.udid),
-            "APP_VER": "1.9.1",
+            "APP_VER": os.environ.get("APP_VER", "1.9.1"), # in case of sent
```
Comment (summertriangle-dev) [inline starlight/apiclient.py]: Should be `VC_APP_VER`

**Comment by CaiMiao** [comment]
Thanks for correcting.

File: starlight/__init__.py [hunk 3]
```diff
@@ -207,6 +207,7 @@ def cache_chars(self, idl):
             kanji_spaced = lambda obj: self.names.get(obj.chara_id).kanji_spaced,
             kana_spaced = lambda obj:  self.names.get(obj.chara_id).kana_spaced,
             conventional =lambda obj: self.names.get(obj.chara_id).conventional,
+            translated =lambda obj: self.names.get(obj.chara_id).translated,
```
Comment (summertriangle-dev) [inline starlight/__init__.py]: bad merge?

File: starlight/__init__.py [hunk 4]
```diff
@@ -377,7 +378,7 @@ def check_version_api_recv(response, msg):
 
     res_ver = msg.get(b"data_headers", {}).get(b"required_res_ver", b"-1").decode("utf8")
     if not data or res_ver != data.version:
-        if res_ver != "-1":
+        if res_ver != -1:
```
Comment (summertriangle-dev) [inline starlight/__init__.py]: ditto

**Comment by summertriangle-dev** [comment]
Thanks for the PR

**Pull request merged**